[Enhancement] remove the speed limitation for upload/download #24683

srlch opened the pull request:
Fixes #24683

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[1094]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|1|00.00%|[1094]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

srlch committed:
[Enhancement] remove the speed limitation for upload/download #24683

Problem:
Two problem which limit the download/upload speed:
1. If we use new authentication framework for S3, the client max connection is 25 by default and is not configurable.
2. backup/restore's download/upload task is limited that be can handle at most 3 task in a single backup/restore task.

Solution:
1. Config max connection by config::object_storage_max_connection.
2. introduce max_upload_task_per_be/max_download_task_per_be to control the max task number for download/upload

Signed-off-by: srlch <linzichao@starrocks.com>

chaoyli commented on be/src/fs/fs_s3.cpp:
```diff
@@ -177,6 +177,7 @@ S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const TCloudConfigurati
     if (!aws_cloud_credential.endpoint.empty()) {
         config.endpointOverride = aws_cloud_credential.endpoint;
     }
+    config.maxConnections = config::object_storage_max_connection;
```
Does it have an effect on the data lake reader?

srlch replied:
It has no side effect on the data lake reader.

dengliu replied:
how can we configure the concurrency for the data sync in share-data mode architecture?
cc @kevincai

kevincai replied:
it is a different io framework for shared data, however most of the configuration items are similar, possibly expose similar configurations in that case.

dengliu replied:
> possibly expose similar configurations in that case.

Are you going to cut a PR for that? We want to harness the most from the AWS network bandwidth and S3

kevincai replied:
will take a look. :)

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 8 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|3|00.00%|[1093, 1094, 1095]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|2|3|66.67%|[548]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

dengliu commented:
thanks for the fix!
can u backport to 3.0?

srlch commented:
> thanks for the fix!
> can u backport to 3.0?

yes of course, I will backport to 3.0 as soon as possible.

yangrong688 commented:
please also backport this pr to branch 2.5, thanks.

srlch commented:
> please also backport this pr to branch 2.5, thanks.

2.5 wil also be backport
